eZ/Publish/Core/FieldType/Date/Value.php

```diff
@@ -80,6 +82,18 @@ public static function fromTimestamp($timestamp)
         }
     }
 
+    /**
+     * Creates a Value from the given DateTimeInterface instance.
+     */
+    public static function fromDateTimeInterface(?DateTimeInterface $dateTime): self
```

Comment:
Shouldn't this argument and a second occurance in this PR similar to this be non-nullable?

Reply:
\`\_\_construct\` will handle \`null\` :wink:

Reply:
@adamwojs Then it's not consistent with `eZ\Publish\Core\FieldType\Time\Value`, which does not accept `null` (while behaving the same I believe).

Reply:
>> `__construct` will handle `null`

> Then it's not consistent with `eZ\Publish\Core\FieldType\Time\Value`, which does not accept `null` (while behaving the same I believe).

I'd probably go for consistency. It kinda doesn't make a sense to name a method "from something" but then not to pass this something.

BTW could you add parent constructor call to these classes while at it? Saw it on EA extended report just now.